zeeguu_api/api/dashboard.py

```diff
@@ -0,0 +1,297 @@
+from datetime import datetime
+
+import flask
+from flask import request
+
+from .utils.json_result import json_result
+from .utils.route_wrappers import cross_domain, with_session
+from . import api
+import zeeguu
+from zeeguu.model import User, Cohort, UserActivityData, Session, Language
+import sqlalchemy
+from flask import jsonify
+import json
+import random
+import datetime
+
+db = zeeguu.db
+
+
+# Checked to see if the user requesting has permission to view the class with id 'class_id'
+def _has_permission_for_class(class_id):
+    from zeeguu.model import TeacherCohortMap
+    link = TeacherCohortMap.query.filter_by(cohort_id=class_id).all()
+    for l in link:
+        if l.user_id == flask.g.user.id:
+            return True
+    return False
+
+
+# Checks to see if user has a valid active session
+@api.route("/has_session", methods=["GET"])
+def has_session():
+    try:
+        session_id = int(flask.request.args['session'])
+        session = Session.query.filter_by(id=session_id).one()
+        return jsonify(1)
+    except:
+        return jsonify(0)
+
+
+@api.route("/get_class_permissions/<id>", methods=["GET"])
+@with_session
+def get_class_permissions(id):
+    """
+
+        Checks to see if user has permissions to access a certain class.
+
+    """
+    if (_has_permission_for_class(id)):
+        return jsonify(1)
+    return jsonify(0)
+
+
+@api.route("/get_user_permissions/<id>", methods=["GET"])
+@with_session
+def get_user_permissions(id):
+    """
+
+        Checks to see if user has permissions to access a certain user.
+
+    """
+    try:
+        user = User.query.filter_by(id=id).one()
+        return get_class_permissions(user.cohort_id)
+    except:
+        flask.abort(401)
+
+
+# Asking for a nonexistant cohort will cause .one() to crash!
+# Takes cohort_id and returns all users belonging to that cohort
+@api.route("/get_users_from_class/<id>", methods=["GET"])
+@with_session
+def get_users_from_class(id):
+    if (not _has_permission_for_class(id)):
+        flask.abort(401)
+    try:
+        c = Cohort.query.filter_by(id=id).one()
+        users = User.query.filter_by(cohort_id=c.id).all()
+        users_info = []
+        for u in users:
+            info = _get_user_info(u.id)
+            users_info.append(info)
+        return json.dumps(users_info)
+
+    except:
+        flask.abort(400)
+
+
+# Takes cohort_id and reuturns dictionary with relevant class variables
+@api.route("/get_user_info/<id>", methods=["GET"])
```

Comment:
i'm puzzled here. 
the comment contradicts the code.
why does it say takes cohort_id?